workflow-ui/core-common/src/main/java/com/squareup/workflow1/visual/VisualFactory.kt

```diff
@@ -23,21 +23,25 @@ public interface VisualFactory<ContextT, in RenderingT, out VisualT> {
    *
    * This method must not call [VisualHolder.update], to ensure that callers have
    * complete control over the lifecycle of the new [VisualT].
+   *
+   * @param getFactory can be used to make recursive calls to build VisualT
+   * instances for sub-parts of [rendering]
    */
   public fun createOrNull(
     rendering: RenderingT,
     context: ContextT,
-    environment: VisualEnvironment
+    environment: VisualEnvironment,
+    getFactory: (VisualEnvironment) -> VisualFactory<ContextT, Any, VisualT>
```

Comment:
This looks redundant:
I know, one thing is the environment and another how to get the factory.
But you could also provide () -> VF...
Anyway I have another question that might affect this, so let's wait for that.

Reply:
We need this in order for `WithEnvironment` to work. `WithEnvironment` needs to be able to redefine what might get returned by that function, that's its main purpose.

Reply:
Hmm. I think? Not going to rathole on this until we discuss your main question.